Framework/API/src/ExperimentInfo.cpp

```diff
@@ -141,14 +146,28 @@ const std::string ExperimentInfo::toString() const {
 */
 void ExperimentInfo::setInstrument(const Instrument_const_sptr &instr) {
   invalidateInstrumentReferences();
-  m_detectorInfo = nullptr;
+  m_detectorInfoWrapper = nullptr;
   if (instr->isParametrized()) {
```

Comment:
Any chance we could make the logic here a little more readable. Maybe extract it to a dedicated function?